scripts/json_overview_image_info.py

```diff
@@ -14,23 +14,37 @@
 
 assert getenv("WORK_DIR"), "$WORK_DIR required"
 
-work_dir = Path(getenv("WORK_DIR"))
-
-output = {}
-
-for json_file in work_dir.glob("*.json"):
-    image_info = json.loads(json_file.read_text())
-    if not output:
-        output.update(image_info)
-    else:
-        # get first (and only) profile in json file
-        device_id = next(iter(image_info["profiles"].keys()))
-        if device_id not in output["profiles"]:
-            output["profiles"].update(image_info["profiles"])
-        else:
-            output["profiles"][device_id]["images"].append(
-                image_info["profiles"][device_id]["images"][0]
-            )
+# merge profiles
+def get_output(work_dir):
+    output = None
+
+    # preserve existing profiles.json file data
+    if output_path.is_file():
+        output = json.loads(output_path.read_text())
+
+    for json_file in work_dir.glob("*.json"):
+        image_info = json.loads(json_file.read_text())
+
+        # use first json file as template
+        if not output:
+            output = image_info
+
+        # get first (and probably only) profile in json file
+        for device_id, profile in image_info["profiles"].items():
+            output["profiles"][device_id] = {
+                **profile,
+                **output["profiles"].get(device_id, {}),
+            }
+            output["profiles"][device_id]["images"].extend(profile["images"])
+
+    # make image lists unique by name
+    for device_id, profile in output["profiles"].items():
+        profile["images"] = list({e["name"]: e for e in profile["images"]}.values())
+
```

Comment:
@aparcar .. and also make the list here unique. (for the image builder)

Reply:
Can you please provide an example where it's not unique? If it would be, the openwrt buildsystem would just overwrite the file itself.

Reply:
I might be wrong here. :)